app/controllers/application_controller.rb

```diff
@@ -5,7 +5,7 @@ class ApplicationController < ActionController::Base
   include SidebarController
 
   before_action :authenticate_user!, unless: :devise_controller?
-  before_action :check_team_presence, unless: :devise_controller?
+  before_action :check_team_presence, if: -> { current_user.present? }
```

Comment:
well, do not discard the previous condition...

In this case you should check both if is a devise_controller and if current_user is present.